### PR TITLE
provider/aws: Ability to reduce nodes in an ElastiCache cluster

### LINKED ISSUE #3604: Not possible to decrease the number of cache nodes in elasticache cluster
We created an AWS Elasticache Memcache cluster with two nodes. We'd now like to change it's configuration to have a single node. 

We get this error while applying the terraform plan to change the num_cache_nodes property from 2 to 1:

Error applying plan:

1 error(s) occurred:

* aws_elasticache_cluster.memcached: [WARN] Error updating ElastiCache cluster (tokyo-staging), error: InvalidParameterCombination: The number of CacheNodeIdsToRemove must match the number of nodes you are trying to remove, current number of cache nodes is 2, requested number of cache nodes is 1 and CacheNodeIdsToRemove supplied are 0
	status code: 400, request id: 75e51204-78f7-11e5-8644-29a7c0ea821f

The relevant AWS docs say:

NumCacheNodes - integer - Optional - The number of Cache Nodes the Cache Cluster should have. If NumCacheNodes is greater than the existing number of Cache Nodes, Cache Nodes will be added. If NumCacheNodes is less than the existing number of Cache Nodes, Cache Nodes will be removed. When removing Cache Nodes, the Ids of the specific Cache Nodes to be removed must be supplied using the CacheNodeIdsToRemove parameter.

CacheNodeIdsToRemove - string|array - Optional - The list of Cache Node IDs to be removed. This parameter is only valid when NumCacheNodes is less than the existing number of Cache Nodes. The number of Cache Node Ids supplied in this parameter must match the difference between the existing number of Cache Nodes in the cluster and the new NumCacheNodes requested. Pass a string for a single value, or an indexed array for multiple values.

I think that terraform should choose an appropriate number of cache nodes ids to remove while applying the plan if the number of nodes is being decremented. For our particular use cases each cache node isn't special, and we use the elasticache configuration endpoint to automatically reconnect if the set changes. 

I don't know how you'd otherwise express which cache nodes to remove? Maybe give the option of providing an array of cache node ids, and remove the nodes in the order provided in the array for users of terraform who care which particular cache nodes are deleted when capacity is decreased.

### PR DESCRIPTION
**Pull request by stack72**
Fixes #3604 

Assuptions:

*We always remove the higher numbered nodes from the cache cluster*

I.e. If we are changing from a 2 node cluster to a 1 node cluster then we remove node_id 0002
If we are changing from a 5 node cluster to a 3 node cluster then we remove node_ids 0005 and 0004

### CODE REVIEW

File: builtin/providers/aws/resource_aws_elasticache_cluster_test.go [hunk 1]
```diff
@@ -173,13 +208,11 @@ resource "aws_security_group" "bar" {
         cidr_blocks = ["0.0.0.0/0"]
     }
 }
-
```
Comment (jen20) [inline builtin/providers/aws/resource_aws_elasticache_cluster_test.go]: There's a decent amount of whitespace noise here, could you revisit that?

**Comment by jen20** [comment]
Hi @stack72, other than the minor whitespace issue this looks good to me. While fixing that, if you want to squash the commits that would be good as well :-)

**Comment by stack72** [comment]
@jen20 done and done :)

**Comment by jen20** [comment]
LGTM!

**Comment by jen20** [comment]
I think we should drop a note into the docs as well pointing out the strategy used when reducing the number of nodes - I'll land a PR for that now.

**Pull request merged**